src/databricks/labs/ucx/azure/access.py

```diff
@@ -183,63 +188,98 @@ def _policy_config(value: str):
+                "{{" + principal_secret_identifier + "}}",
+            ),
+        ]
+
     def _update_sql_dac_with_spn(
         self,
-        storage_account_info: list[StorageAccount],
-        uber_principal: PrincipalSecret,
-        inventory_database: str,
+        principal_client_id: str,
+        principal_secret_identifier: str,
+        storage_accounts: list[StorageAccount],
     ):
-
         warehouse_config = self._ws.warehouses.get_workspace_warehouse_config()
-        sql_dac = warehouse_config.data_access_config
-        if sql_dac is None:
-            sql_dac = []
-        tenant_id = self._azurerm.tenant_id()
-        endpoint = f"https://login.microsoftonline.com/{tenant_id}/oauth2/token"
-        for storage in storage_account_info:
-            sql_dac.extend(
-                [
-                    EndpointConfPair(
-                        f"spark_conf.fs.azure.account.oauth2.client.id.{storage.name}.dfs.core.windows.net",
-                        uber_principal.client.client_id,
-                    ),
-                    EndpointConfPair(
-                        f"spark_conf.fs.azure.account.oauth.provider.type.{storage.name}.dfs.core.windows.net",
-                        "org.apache.hadoop.fs.azurebfs.oauth2.ClientCredsTokenProvider",
-                    ),
-                    EndpointConfPair(
-                        f"spark_conf.fs.azure.account.oauth2.client.endpoint.{storage.name}.dfs.core.windows.net",
-                        endpoint,
-                    ),
-                    EndpointConfPair(
-                        f"spark_conf.fs.azure.account.auth.type.{storage.name}.dfs.core.windows.net", "OAuth"
-                    ),
-                    EndpointConfPair(
-                        f"spark_conf.fs.azure.account.oauth2.client.secret.{storage.name}.dfs.core.windows.net",
-                        "{{secrets/" + inventory_database + "/uber_principal_secret}}",
-                    ),
-                ]
+        self._installation.save(warehouse_config, filename="warehouse-config-backup.json")
+        sql_dac = warehouse_config.data_access_config or []
+        for storage in storage_accounts:
+            configuration_pairs = self._create_storage_account_data_access_configuration_pairs(
+                principal_client_id, principal_secret_identifier, storage
             )
+            sql_dac.extend(configuration_pairs)
+        self._ws.warehouses.set_workspace_warehouse_config(
+            data_access_config=sql_dac,
+            sql_configuration_parameters=warehouse_config.sql_configuration_parameters,
+        )
+
+    def _revert_sql_dac_with_spn(
+        self,
+        principal_client_id: str,
+        principal_secret_identifier: str,
+        storage_accounts: list[StorageAccount],
+    ):
+        try:
+            warehouse_config = self._installation.load(
+                GetWorkspaceWarehouseConfigResponse, filename="warehouse-config-backup.json"
```

Comment:
```suggestion
                GetWorkspaceWarehouseConfigResponse, filename="workspace-warehouse-config-backup.json"
```
might be a bit better name for consistency